stdlib/Printf/src/Printf.jl

```diff
@@ -3,25 +3,306 @@
 __precompile__(true)
 
 module Printf
-# the macro implementations here exactly mirrors the
-# macros left in base/printf.jl, and uses the utility there
+# the implementations here is copied from
+# what is left in base/printf.jl, and uses the utility there
 
-export @printf, @sprintf
-using Base.Printf: _printf, is_str_expr, fix_dec, DIGITS, print_fixed, decode_dec, decode_hex,
+export @printf, @sprintf, @format_str
+export format, format_expr, printf, sprintf
+
+import Base.Printf: is_str_expr, fix_dec, DIGITS, print_fixed, decode_dec, decode_hex,
                    ini_hex, ini_HEX, print_exp_a, decode_0ct, decode_HEX, ini_dec, print_exp_e,
                    decode_oct, _limit
-using Unicode.textwidth
+
+using Base.Printf: gen_f, gen_e, gen_a, gen_g, gen_c, gen_s, gen_p, gen_d
+using Unicode:  lowercase, textwidth
+
+# copied from Base.Printf - amended by additional features:
+# 1. optional argument positions %(\d+[$&])
+# 2. special term %\d+% standing for %\d+&s
```

Comment:
Does this support the same feature set both in the macro and function forms?

Reply:
Yes.